File: codewars/Js/AreSame.js
function comp(array1, array2) {
  if (array1 == null || array2 ==null) {
    return false;
  }

  if (array1.length != array2.length) {
    return false;
  }

  let square = [];
  for (let i = 0; i < array1.length; i++) {
    square[i] = array1[i] * array1[i];
  }

  square.sort();
  array2.sort();

  for (let i = 0; i < square.length; i++) {
    if (square[i]!== array2[i]) {
      return false;
    }
  }
  return true;
}

function Ex2(array1, array2) {

}

File: codewars/Java/AreSame.java
import java.util.Arrays;

public class AreSame {
  public static boolean comp(int[] a, int[] b) {

    if (a == null || b == null) {
      return false;
    }

    if (a.length != b.length) {
      return false;
    }

    int[] squares = Arrays.stream(a)
        .map(x -> x * x)
        .toArray();

    Arrays.sort(b);
    Arrays.sort(squares);

    for (int i = 0; i < a.length; i++) {
      if (squares[i] != b[i]) {
        return false; // 하나라도 다르면 false 반환
      }
    }

    return true;
  }
}

// a 배열 에 수의 제곱이 b 배열에 다 들어있으면 된다.